### PR TITLE
[MetricsAdvisor] Added GetDatasourceCredentials to admin client

### PR DESCRIPTION
**Pull request by kinelski**
Part of https://github.com/azure/azure-sdk-for-net/issues/18609.

Added listing method, `GetDatasourceCredentials`, to the API. Also, after feedback from Python's architects, renamed all instances of `DataSourceCredentialEntity` to `DatasourceCredential`.

### CODE REVIEW

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/DisposableCredentialEntity.cs [hunk 1]
```diff
@@ -39,15 +39,15 @@ private DisposableCredentialEntity(MetricsAdvisorAdministrationClient adminClien
...
+        /// A <see cref="DatasourceCredential"/> instance is returned, from which the ID of the created
         /// entity can be obtained. Upon disposal, the associated entity will be deleted.
         /// </summary>
         /// <param name="adminClient">The client to use for creating and for deleting the credential entity.</param>
-        /// <param name="credentialEntity">Specifies how the created <see cref="DataSourceCredentialEntity"/> should be configured.</param>
+        /// <param name="credentialEntity">Specifies how the created <see cref="DatasourceCredential"/> should be configured.</param>
         /// <returns>A <see cref="DisposableCredentialEntity"/> instance from which the ID of the created entity can be obtained.</returns>
-        public static async Task<DisposableCredentialEntity> CreateCredentialEntityAsync(MetricsAdvisorAdministrationClient adminClient, DataSourceCredentialEntity credentialEntity)
+        public static async Task<DisposableCredentialEntity> CreateCredentialEntityAsync(MetricsAdvisorAdministrationClient adminClient, DatasourceCredential credentialEntity)
```
[the remark this answers is not in the document]
Comment (kinelski) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/DisposableCredentialEntity.cs:48]: I realized I missed this class in the next set of changes I'm working on. Already renamed it there, so it should be fixed in the next PR.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs [hunk 2]
```diff
@@ -243,20 +249,20 @@ public partial class MetricsAdvisorAdministrationClient
...
         public virtual Azure.Response<Azure.AI.MetricsAdvisor.Models.DataFeed> CreateDataFeed(Azure.AI.MetricsAdvisor.Models.DataFeed dataFeed, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual System.Threading.Tasks.Task<Azure.Response<Azure.AI.MetricsAdvisor.Models.DataFeed>> CreateDataFeedAsync(Azure.AI.MetricsAdvisor.Models.DataFeed dataFeed, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Response<Azure.AI.MetricsAdvisor.Models.DatasourceCredential> CreateDatasourceCredential(Azure.AI.MetricsAdvisor.Models.DatasourceCredential datasourceCredential, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
[the remark this answers is not in the document]
Comment (kinelski) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs:254]: Johan wanted us to treat "Datasource" as a single word, so we're aligning across languages for this preview. This hasn't been approved by .NET architects yet.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs [hunk 3]
```diff
@@ -1888,32 +1888,32 @@ private static void ValidateHookToCreate(NotificationHook hook, string paramName
...
-        /// Please see <see cref="DataSourceCredentialEntity"/> for a list of supported credentials.
+        /// Please see <see cref="DatasourceCredential"/> for a list of supported credentials.
         /// </summary>
-        /// <param name="credentialEntity">Specifies how the created <see cref="DataSourceCredentialEntity"/> should be configured.</param>
+        /// <param name="datasourceCredential">Specifies how the created <see cref="DatasourceCredential"/> should be configured.</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
         /// <returns>
-        /// A <see cref="Response{T}"/> containing the result of the operation. The result is a <see cref="DataSourceCredentialEntity"/>
+        /// A <see cref="Response{T}"/> containing the result of the operation. The result is a <see cref="DatasourceCredential"/>
         /// instance containing information about the created credential entity.
```
[the remark this answers is not in the document]
Comment (kinelski) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs:1899]: Addressed locally.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (maririos) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs:254]: interesting.. but it is not a single word.....
yeah I guess, whatever the .NET arch say.
Let me know though, I am curious

**Review by maririos — APPROVED** [review]

**Comment by maririos** [comment]
Super late, but I forgot to say that there was no changelog in this PR

**Pull request merged**